app/models.py

```diff
@@ -85,7 +85,7 @@ class Drink(db.Model):
     highlight = db.Column(db.Boolean)
 
     def getprice(self):
-        return format_curr(self.price)
+        return format_currency(self.price, 'EUR', u'#,##0.00 ¤', locale=app.config['BABEL_DEFAULT_LOCALE'])
```

Comment:
You don't need to set the locale, as this is already determinded in the request context.
The format of the currency amount will be done by babel self.

Reply:
Thanks for pointing that out.